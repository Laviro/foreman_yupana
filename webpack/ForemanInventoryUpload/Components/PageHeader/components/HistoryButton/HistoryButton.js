import React from 'react';
import { Button, Icon } from 'patternfly-react';
import { ACTIONS_HISTORY_BUTTON_TEXT } from '../../../../ForemanInventoryConstants';
import { getActionsHistoryUrl } from '../../../../ForemanInventoryHelpers';

const HistoryButton = () => (
  <Button
    className="tasks_history_button"
    href={getActionsHistoryUrl()}
    target="_blank"
    rel="noopener noreferrer"
  >
    <Icon name="history" />
    {ACTIONS_HISTORY_BUTTON_TEXT}
  </Button>
);

export default HistoryButton;
