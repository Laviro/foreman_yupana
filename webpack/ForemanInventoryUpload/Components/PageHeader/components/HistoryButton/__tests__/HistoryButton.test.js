import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import HistoryButton from '../HistoryButton';

global.URL_PREFIX = '';

const fixtures = {
  'render without Props': {},
};

describe('HistoryButton', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(HistoryButton, fixtures));
});
