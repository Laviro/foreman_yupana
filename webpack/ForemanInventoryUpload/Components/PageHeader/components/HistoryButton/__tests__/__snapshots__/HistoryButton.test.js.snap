// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryButton rendering render without Props 1`] = `
<Button
  active={false}
  block={false}
  bsClass="btn"
  bsStyle="default"
  className="tasks_history_button"
  disabled={false}
  href="/foreman_tasks/tasks?search=action++%3D++ForemanInventoryUpload%3A%3AAsync%3A%3AGenerateReportJob+or+action++%3D++ForemanInventoryUpload%3A%3AAsync%3A%3AGenerateAllReportsJob&page=1"
  rel="noopener noreferrer"
  target="_blank"
>
  <Icon
    name="history"
    type="fa"
  />
   Actions history
</Button>
`;
